### PR TITLE
384119 Set up queries frontend

### PR DESCRIPTION
**Pull request by rtasalem**
https://dev.azure.com/defragovuk/DEFRA-FFC/_workitems/edit/384119

### CODE REVIEW

**Comment by rtasalem** [comment]
After building & starting the container, both endpoints (`locahost:3017/queries/healthy` & `localhost:3017/queries/healthz`) are returning `ok` response as expected.

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ARG PARENT_VERSION=2.1.2-node18.11.0
-ARG PORT=3000
-ARG PORT_DEBUG=9229
```
Comment (johnwatson484) [inline Dockerfile:3]: Should be using LTS Node version 20.

File: docker-compose.yaml [hunk 2]
```diff
@@ -8,3 +8,7 @@ services:
     environment:
       APPINSIGHTS_CLOUDROLE: ffc-sfd-queries-local
       APPINSIGHTS_CONNECTIONSTRING: ${APPINSIGHTS_CONNECTIONSTRING}
+      AUTH_HOST: ${AUTH_HOST:-http://ffc-sfd-proxy:3000}
+      GATEWAY_HOST: ${GATEWAY_HOST:-http://localhost:3000}
+      DATA_HOST: ${DATA_HOST:-http://ffc-sfd-data:3004/graphql}
+      AHWP_HOST: ${AHWP_HOST:-http://localhost:3010}
```
Comment (johnwatson484) [inline docker-compose.yaml]: Don't think this service needs to know anything about the AHWP stub service?

File: package.json [hunk 3]
```diff
@@ -10,42 +10,47 @@
...
+  ],
   "license": "OGL-UK-3.0",
   "dependencies": {
+    "@hapi/crumb": "^9.0.1",
     "@hapi/hapi": "21.3.2",
     "@hapi/inert": "7.1.0",
     "@hapi/vision": "7.0.3",
     "applicationinsights": "2.8.0",
-    "govuk-frontend": "4.7.0",
+    "ffc-auth": "^0.12.0",
```
Comment (johnwatson484) [inline package.json]: Need to drop the `^` in this file

Comment (johnwatson484) [reply package.json]: You can update your default npm config to not set them by default

**Review by johnwatson484 — CHANGES_REQUESTED** [review]
Needs a tweak to the export on `server.js` as `createServer` needs wrapping in `{ }`

Currently app fails to start up

File: Dockerfile (hunk 1 above)
Comment (rtasalem) [reply Dockerfile:3]: Updated the version to v20.

File: docker-compose.yaml (hunk 2 above)
Comment (rtasalem) [reply docker-compose.yaml]: Ah thank you, have removed this reference.

File: package.json (hunk 3 above)
Comment (rtasalem) [reply package.json]: Done.

**Review by johnwatson484 — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (rtasalem) [reply package.json]: > You can update your default npm config to not set them by default

I thought I had but realised I used the wrong command. Have now updated my NPM config so the caret (`^`) is no longer there when I install packages.

**Pull request merged**